### PR TITLE
Finalize Trader

### PR DESCRIPTION
**Pull request by martinkersner**

### CODE REVIEW

File: seoulai_gym/envs/traders/base.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+"""
+Cinyoung Hur, cinyoung.hur@gmail.com
+seoulai.com
+2018
+"""
+
+
+class Constants(object):
+    """ Constants to share between classes and functions for checkers game.
```
Comment (martinkersner) [inline seoulai_gym/envs/traders/base.py]: checkers game?

File: README.md [hunk 2]
```diff
@@ -70,7 +70,7 @@ pip3 install -e .
 `seoulai-gym` requires to have at least Python 3.6 and was tested on Arch Linux, macOS High Sierra and Windows 10.
 
 ## Environments
-Currently, environment simulating game of [Checkers](https://en.wikipedia.org/wiki/Draughts) and [Mighty](https://en.wikipedia.org/wiki/Mighty_(card_game)) are provided.
+Currently, environment simulating game of [Checkers](https://en.wikipedia.org/wiki/Draughts), [Mighty] (https://en.wikipedia.org/wiki/Mighty_(card_game)), and Trader are provided.
```
Comment (martinkersner) [inline README.md]: Sometimes you call it `Trader`, sometimes you call it `Market`. I feel little bit confused.

File: seoulai_gym/envs/traders/agents.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        self.asset_val = 0.0
+
+    def act(
+        self,
+        obs,  # price history
+        reward: int,
+        done: bool,
+    ) -> Tuple[int, int, int]:
+        """
+        총 자산 = 현금 잔고 + 주식 가격 * 주식 수량
```
Comment (martinkersner) [inline seoulai_gym/envs/traders/agents.py]: It would be great if comments are in English :)

File: examples/trader_example.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+Cinyoung Hur, cinyoung.hur@gmail.com
+James Park, laplacian.k@gmail.com
+seoulai.com
+2018
+"""
+import seoulai_gym as gym
+from seoulai_gym.envs.traders.agents import RandomAgentBuffett
+
+
+def main():
```
Comment (martinkersner) [inline examples/trader_example.py]: Example code does not seem to work well for me (on MacOS, I will try Linux later) or I am not sure what it should do :)

Comment (hurcy) [reply examples/trader_example.py]: There was a matplotlib issue. James is working on this.

**Comment by martinkersner** [comment]
@hurcy @laplacian-k On macOS trader does not display me anything. On Linux it throws `[1]    20242 segmentation fault (core dumped)  python examples/trader_example.py` at the end. Did you have similar issue?

File: seoulai_gym/envs/traders/market.py [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+        self,
+        agent,
+        decision,
+        trad_price: float,
+        trad_qty: int,
+    ):
+        """Make a step (= move) within market.
+        Args:
+            agent: Agent making a move.
+            from_row: Row of board of original piece location.
```
Comment (martinkersner) [inline seoulai_gym/envs/traders/market.py]: This is comment for checkers.

**Comment by hurcy** [comment]
I’m aware of it. :-) I assume segmentation fault is caused by matplotlib.

**Comment by martinkersner** [comment]
@hurcy Are you planning to do something with it or ignore it for now?

File: seoulai_gym/envs/traders/market.py (hunk 5 above)
Comment (laplacian-k) [reply seoulai_gym/envs/traders/market.py]: I just fixed and push it.

**Comment by laplacian-k** [comment]
There are three ways to fix matplotlib error in macOS.

1-1. change import structure of graphics.py
+ import matplotlib
+ matplotlib.use('Agg')
import matplotlib.backends.backend_agg as agg
import matplotlib.pyplot as plt
import pylab
import pygame

but, It doesn't satisfy 'flake8' condition.

1-2. hurcy's method.
I heard that hurcy added import matplotlib and matplotlib.use('Agg') in __init__.py in order to satisfy 'flake8' condition. Does not this work?

2. execute below command in your bash environment.
echo "backend: Agg" >> ~/.matplotlib/matplotlibrc
It does work well!
but, I don't know the way to apply this command to all seoulai gym users.
Can we automate this script?

**Comment by hurcy** [comment]
@martinkersner I tested on my laptop, desktop(both Ubuntu 16.04). I don't see seg fault. But I need to check in other platforms, so I'll working on it with James.

File: seoulai_gym/envs/traders/market.py (hunk 5 above)
Comment (martinkersner) [reply seoulai_gym/envs/traders/market.py]: @laplacian-k where did you push?

Comment (laplacian-k) [reply seoulai_gym/envs/traders/market.py]: Sorry, I pushed in laplacian-k/gym yesterday.
And I just git clone trader branch. then, I tried to push fixed market.py file in seoulai/gym, trader branch. but I couldn't push that file

> git push origin trader
> remote: Permission to seoulai/gym.git denied to laplacian-k.
> ...
How can I do?

Comment (martinkersner) [reply seoulai_gym/envs/traders/market.py]: @laplacian-k let me check it

Comment (martinkersner) [reply seoulai_gym/envs/traders/market.py]: @laplacian-k 
Try now!

**Comment by martinkersner** [comment]
@hurcy @laplacian-k I will merge it. I think it is okay for now. Segmentation fault is maybe because of Python3.7 and problem with macOS can be solved later. From now on, make always branch directly at `gym` repository and then PR to merge to master.

Thank you for your contribution!

**Pull request merged**